File: Blind1.java
import java.util.HashMap;
//this solution utilizes a hashmap
public class Blind1 {
    public static int[] twoSum(int[] nums, int target) {
        //create a new hashmap that can hold a key value pair of integers
        //keys being the element's integer value and value being the element's index
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        //run through the input array and perform functions
        for(int i = 0; i < nums.length; i++){
            //the required number is the complement of the target and the current element in the loop
            Integer complement = (Integer)(target - nums[i]);
            //if the hashmap contains the complement which when added to the current element equals
            //the target, it will store the complement's index and the current element's index into 
            //the desired return type of an int array
            if(map.containsKey(complement)){
                int solution[] = {map.get(complement), i};
                return solution;
            }
            //if the complement is not found, the current key value pair is stored into the hashmap
            //and the process will repeat with the next element of the input array
            map.put(nums[i], i);
        }
        //return null statement just in case ;)
        return null;    
    }
}
//this solution is pretty nice because the time complexity is O(n) since in the worst case scenario, 
//we would have to iterate through the entire input array. 
// -- JAVA SKILLS USED --
//Hashmap initialization: in line 7 to create a hashmap with key value pairs of Integers
//Type casting: in line 11 because the Hashmap deals with the type Integer not int
//Hashmap.containsKey(): in line 15 finds the desired key (the element's value in our case) in the map
//Hashmap.get(): in line 16 gets the value of the map that matches with the key
//Hashmap.put(): in line 21 stores a key value pair into the map

File: Blind1.py
def two_sum(nums: list[int], target: int) -> list[int]:
    #create an empty dictionary (same as key value hashmap in java)
    dict = {}
    #enumerate nums to make a list of key value pairs of element index and element value 
    for i, n in enumerate(nums):
        #the complement is the number we need to get the solution with this particular element
        complement = target - n
        #if the complement is in the dictionary as a value, set a variable to the index of the input
        #array by calling the element index of the complement by its key (the element value)
        # then return the solution
        if complement in dict:
            j = dict[complement]
            return [i, j]
        #if the complement is not yet in our dictionary, then we add the current index, value pair to it
        #using the element value as the key and the element index as the value (dont be confused by the names)
        else:
            dict[n] = i
        #after adding to our dictionary, repeat the process again with the next element in the input array

#This solution is pretty much the same as the Java one and also has runtime O(n). It effectively uses
#the enumerate function in python to make the input easy to use!
# -- PYTHON SKILLS USED --
#Creating an empty dictionary: in line 3 
#enumerate(): in line 5 to split the input array into index, value pairs
#calling a dictionary value by its key: in line 12. rememeber, the key is the element's value and the value is the element's index
#adding to a dictionary: in line 17. you can add to a dictionary in the form dict[key] = value